### PR TITLE
Fix more generic for platforms that do not have execinfo.h

### PR DESCRIPTION
**Pull request by simbit18**
## Summary
Simulator build fails on non-GLIBC platform #5621 #5623.
On platforms where the execinfo.h file is missing, the Gnulib substitute implementation is just a stub, and does nothing.
https://www.gnu.org/software/gnulib/manual/html_node/execinfo_002eh.html

## Impact
On Linux with glibc: There should be no impact.
On other platforms: You need to add SIM_GLIBC_PLATFORM=n to your .config file to enable simulator compilation.

## Testing
We tested on platform Alpine linux [sim:nsh on github](https://github.com/simbit18/nuttx-alpine)  sim:lua

### CODE REVIEW

File: arch/sim/src/sim/posix/sim_hostmisc.c [hunk 1]
```diff
@@ -90,10 +90,10 @@ void host_abort(int status)
 
 int host_backtrace(void** array, int size)
 {
-#ifdef CONFIG_WINDOWS_CYGWIN
-  return 0;
-#else
+#ifdef CONFIG_SIM_GLIBC_PLATFORM
```
[the remark this answers is not in the document]
Comment (acassis) [reply arch/sim/src/sim/posix/sim_hostmisc.c:93]: I think __has_include is not portable. You need to test first if it is defined.

https://gcc.gnu.org/onlinedocs/cpp/_005f_005fhas_005finclude.html

[the remark this answers is not in the document]
Comment (xiaoxiang781216) [reply arch/sim/src/sim/posix/sim_hostmisc.c:93]: __has_include is part of c++17 spec, should become c spec soon, but it's always good to check defined(__has_include) first.

[the remark this answers is not in the document]
Comment (simbit18) [reply arch/sim/src/sim/posix/sim_hostmisc.c:93]: #if defined __has_include
#   if __has_include(<execinfo.h>)
#      define SIM_GLIBC_PLATFORM 1
#   endif
#endif
I tested on Alpine and Mingw it's ok. I will update this PR.

**Comment by simbit18** [comment]
Sorry, there was a mistake in this change.

**Pull request closed without merging**